### ISSUE TITLE
[camelcase] Allow disabling camelcase rule for class members

### ISSUE THREAD

**Issue by jimjenkins5** [typescript-eslint/typescript-eslint#590]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/camelcase": ["error", { "properties": "never" } ]
  }
}
```

```TS
export class MyClassWithAStatic {
  public static this_should_pass(): boolean {
    return true;
  }
}
```

**Expected Result**

The static method should pass linting.

**Actual Result**

An error is reported:
```bash
2:18  error  Identifier 'this_should_pass' is not in camel case  @typescript-eslint/camelcase
```

**Additional Info**
When disabling the camelCase rule for properties, I would assume class members would also not be checked.  However, ideally we would have even more control over this rule and be able to disable it for certain types of members (e.g. by accessibility, and especially for our use, static vs. instance)  

The reason this is causing us an issue is that our coding standard is to use camelCase for all variables, properties and methods, but use snake_case for static methods.  This is a common coding standard and it would be good to support it. 

Even if our use case isn't specifically supported, being able to disable the rule for class methods and properties would be consistent with the current `properties` option.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.2.0` |
| `TypeScript`                       | `3.3.1` |
| `ESLint`                           | `5.11.1` |
| `node`                             | `8.10.0` |
| `npm`                              | `6.4.1` |

**Comment by moulikcipherX** [comment]
@jimjenkins5 does static members also support snake_case?

**Comment by golopot** [comment]
Since this rule is extended from the core rule with the same name, and that the proposed change is not specific to typescript, maybe this change should be proposed in eslint core?

**Comment by bradzacher** [comment]
That's a good point - this particular error has nothing to do with typescript.
Though to be fair, technically a class method is not a property.
There is no option for it in the base rule.

This issue should be raised in the core repo.

There is the other half of the problem; eslint core doesn't support class properties, which we can patch in in our rule.
